addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -2462,6 +2462,29 @@ X[]
                     contentAfter: '<p>a[]l</p>',
                 });
             });
+            it('should remove a table if completely selected', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: unformat(
+                        `<table><tbody>
+                        <tr><td>[a</td><td>b</td><td>c</td></tr>
+                        <tr><td>d</td><td>e</td><td>f</td></tr>
+                        <tr><td>g</td><td>h</td><td>i</td></tr>
+                        </tbody></table>
+                        <p>]<br></p>`
+                    ),
+                    stepFunction: async editor => {
+                        await deleteBackward(editor);
+                        const sel = editor.document.getSelection();
+                        const range = editor.document.createRange();
+                        range.setStart(editor.editable, 0);
+                        range.setEnd(editor.editable, 0);
+                        sel.removeAllRanges(range);
+                        sel.addRange(range);
+                        getDeepRange(editor.editable, {range, select: true});
```

Comment:
@deso-odoo , we actually don't need this,`getDeepRange(editor.editable)` will do the job for us. Can you give it a try?